tdvt/tdvt/config_gen/datasource_list.py

```diff
@@ -356,14 +356,16 @@ def load_test(config, test_dir=get_root_dir()):
                 logging.debug(e)
                 pass
         # Add smoke tests
-        elif connection_test in section:
+        elif sect.get('StaplesSmokeTestEnabled') == 'True' or sect.get('CalcsSmokeTestEnabled') == 'True':
```

Comment:
This doesn't run any tests for a ds with smoke tests enabled. Can you confirm that this runs on your local?
<img width="1232" alt="2023-07-12_13-57-15" src="https://github.com/tableau/connector-plugin-sdk/assets/5643640/4ec2ffa6-c9bc-40d9-827d-98f00f8727fc">
```[ConnectionTests]
StaplesSmokeTestEnabled=True
CalcsSmokeTestEnabled=True
```
From the error trace on the original bug: 
```File "C:\Python310\lib\site-packages\tdvt\test_results.py", line 227, in return_testcaseresult_for_not_run_tests
return TestCaseResult('', str(test_case_count), "", test_case_count, '',
```
 I think you need to look at `test_results.py:227`, in the `else` block on line 229 my linter says that the `TestCaseResult` object that's being instantiated is missing the `test_metadata` bit there. 
<img width="1101" alt="Screen Shot 2023-07-12 at 2 00 21 PM" src="https://github.com/tableau/connector-plugin-sdk/assets/5643640/ca844fb7-49d6-44e1-94d1-af976fb09c3a">

Reply:
[This helper function](https://github.com/tableau/connector-plugin-sdk/pull/1151/files#diff-c3fb5ac7f4637a097c02f6ae6ffe673f779c4e10b318bb0e55ba50b70b4f33ccR84) is where potential bad logic is.